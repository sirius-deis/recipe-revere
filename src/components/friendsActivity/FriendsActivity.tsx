import { forwardRef } from "react";
import styles from "./FriendsActivity.module.css";
import { Link } from "react-router-dom";

interface FriendsActivityProps {
  _id: string;
  name: string;
  picture: string;
  activity: string;
  time: number;
}

const FriendsActivity = forwardRef<HTMLDivElement, FriendsActivityProps>(
  ({ _id, name, picture, activity, time }, ref) => {
    return (
      <article ref={ref}>
        <div className={styles["image-container"]}>
          <img src={picture} alt={name} />
        </div>
        <div className={styles.activity}>
          <h3>
            <Link to={`/users/${_id}`}>{name}</Link>
          </h3>
          just
          <p className={styles.action}>{activity}</p>
        </div>
        <p className={styles.time}>{time}</p>
      </article>
    );
  }
);

export default FriendsActivity;
