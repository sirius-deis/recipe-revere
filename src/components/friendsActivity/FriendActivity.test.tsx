import { render, screen } from "@testing-library/react";
import { BrowserRouter } from "react-router-dom";
import FriendsActivity from "./FriendsActivity";

const time = new Date("2023-12-15T14:16:13").getTime();

const dummy = {
  _id: "123",
  name: "friend name",
  picture: "picture",
  activity: "some activity",
  time: time,
};

describe("FriendsActivity component", () => {
  it("should match snapshot", () => {
    const { container } = render(
      <BrowserRouter>
        <FriendsActivity {...dummy} />
      </BrowserRouter>
    );
    expect(container).toMatchSnapshot();
  });
  it("should render props correctly", () => {
    render(
      <BrowserRouter>
        <FriendsActivity {...dummy} />
      </BrowserRouter>
    );
    expect(screen.getByText("friend name")).toBeInTheDocument();
    expect(screen.getByText("some activity")).toBeInTheDocument();
    const img = screen.getByRole("img");
    expect(img).toBeInTheDocument();
    expect(img).toHaveAttribute("src", "picture");
    expect(img).toHaveAttribute("alt", "friend name");
    expect(screen.getByRole("link")).toHaveAttribute("href", "/users/123");
    expect(screen.getByText(new Date(time).toISOString())).toBeInTheDocument();
  });
});
