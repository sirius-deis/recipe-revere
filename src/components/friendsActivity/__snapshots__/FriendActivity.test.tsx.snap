// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FriendsActivity component should match snapshot 1`] = `
<div>
  <article
    class="friends-activity"
  >
    <div
      class="image-container"
    >
      <img
        alt="friend name"
        src="picture"
      />
    </div>
    <div
      class="activity"
    >
      <h3
        class="name"
      >
        <a
          href="/users/123"
        >
          friend name
        </a>
      </h3>
      just
      <p
        class="action"
      >
        some activity
      </p>
    </div>
    <p
      class="time"
    >
      2023-12-15T13:16:13.000Z
    </p>
  </article>
</div>
`;
